src/contract/event.rs

```diff
@@ -229,6 +237,14 @@ impl<T: Transport, E: ParseLog> AllEventsBuilder<T, E> {
         self
     }
 
+    /// Limit the number of events that can be retrieved by this filter.
+    ///
+    /// Note that this parameter is non-standard.
+    pub fn limit(mut self, value: usize) -> Self {
+        self.filter = self.filter.limit(value);
+        self
+    }
```

Comment:
I see this snippet in more than 1 place... Is this intended?

Reply:
> I only wonder why was it ever removed in the first place?

It was removed as part of a refactor, with intention of adding it back later.

> Is this intended?

Yes, there are more than one kind of builder that can set the `limit` property. These methods are just small helpers for setting that property.